markdown/blog/asyncapi-and-websocket.md

```diff
@@ -0,0 +1,581 @@
+
+If I attempt to parse a message that isn't included in the list of messages for the #chat channel, as shown below...
+
+```
+operations:
+  sendMessage:
+    summary: Send a chat message
+    description: Allows users to send messages to the chat room
+    action: send
+    channel:
+      $ref: '#/channels/chat'
+    messages:
+      - $ref: '#/channels/chat/messages/hello'
+```
+
+This will fail because in my #chat channel, i have no such message as `hello` even if i have the `hello` message defined in my message component. 
+
+A good thing to keep at the back of your mind when defining an operation is the list of messages you're assigning to an operation has to be available in the linked channel messages. 
+
+
+Now that we've created our first operation that allows us to send message, we also need to create another operation that allows us to receive a message. And we do that by doing almost same thing as sending a message except, instead of `send` in the action, we use the `receive` action, just as seen below.
+
+```
+operations:
+  sendMessage:
+    summary: Receive a chat message
+    description: Allows users to receive messages to the chat room
+    action: receieve
+    channel:
+      $ref: '#/channels/chat'
+    messages:
+      - $ref: '#/channels/chat/messages/chatMessage'
+```
+
+With this implementation, we have a fully functional AsyncAPI document, but want to go a few more steps
+
+### Step 6 - Reusing an Existing Message Component
+
+Let’s say we want our server to notify users whenever someone joins or leaves the chat. How would we approach this?
+
+First, we define the new message in our components section. This message will hold information about the user joining or leaving.
+
+```
+components:
+  messages:
+   chat:
+     ...
+   status:
+     payload:
+       type: object
+       properties:
+         userId:
+           type: string
+           description: ID of the user that joined or left
+         type:
+           type: string
+           enum:
+             - join
+             - leave
+         username:
+           type: string
+           description: Display name of the user
+         timestamp:
+           type: string
+           format: date-time
+         required:
+           - userId
+           - username
+           - timestamp
+     
+```
+Here, we’ve created a new `status` message to capture details about users joining or leaving.
+
+Next, let’s add this message to our channel, so our server can broadcast it as needed:
+
+```
+channels:
+  chat:
+    address: /
+    title: Users channel
+     messages:
+       chatMessage:
+         $ref: '#/components/messages/chat'
+         
+       userStatus: #newly added channel message
```

Comment:
instead of adding comment you can use `CodeBlock` component like https://github.com/asyncapi/website/blob/master/markdown/docs/tutorials/getting-started/servers.md?plain=1#L14

Reply:
I don't think the `CodeBlock` highlight is working as expected. I thought it was suppose to highlight the lines specified in the props 🤔